enumerables.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array
+  end
+
+  def my_all?
+    selfitem = self
+    selfitem.my_each { |x| return false unless yield(x) }
+    true
+  end
+
+  def my_any?
+    selfitem = self
+    selfitem.my_each { |x| return true if yield(x) }
+    false
+  end
+
+  def my_none?
+    selfitem = self
+    selfitem.my_each { |x| return false if yield(x) }
+    true
+  end
```

Comment:
enumerables my_none? when no block or argument is given returns true only if none of the collection members is true
      Failure/Error: expect(false_array.my_none?).to be true
      
      LocalJumpError:
        no block given (yield)

 enumerables my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be true
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

enumerables my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

enumerables my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)